modules/core/include/opencv2/core/utils/logger.defines.hpp

```diff
@@ -17,6 +17,26 @@
 #define CV_LOG_LEVEL_DEBUG 5           //!< Debug message. Disabled in the "Release" build.
 #define CV_LOG_LEVEL_VERBOSE 6         //!< Verbose (trace) messages. Requires verbosity level. Disabled in the "Release" build.
 
+namespace cv {
+namespace utils {
+namespace logging {
+
+//! Supported logging levels and their semantic
+enum LogLevel {
```

Comment:
Several other header files (log tag manager and log config parser) need to see the enum definition `LogLevel`. Because `LogLevel` is a traditional (non-strongly-typed) enum, and because some code refers to the members of LogLevel (the individual values), they need to include a header that contains the enum, but not the remainder of the content of logger.hpp (to avoid circular header dependency). Therefore, the enum definition `LogLevel` and its members are moved to `logger.defines.hpp` which is a convenient choice because it only contains "defines" (and by a stretch, the definition of a enum), and nothing else, which solves the difficulty described above.